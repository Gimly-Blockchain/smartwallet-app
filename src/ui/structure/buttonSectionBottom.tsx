import React from 'react'
import { StyleSheet, View } from 'react-native'
import { Button } from 'react-native-material-ui'
import { Colors, Buttons, Spacing } from 'src/styles'

interface Props {
  confirmText: string
  denyText: string
  handleConfirm: () => void
  handleDeny: () => void
  disabled: boolean
  denyDisabled?: boolean
  verticalPadding?: number
}

export const ButtonSection: React.FC<Props> = props => {
  const styles = StyleSheet.create({
    buttonBlock: {
      flex: 1,
      borderColor: Colors.lightGrey,
      borderTopWidth: 1,
      backgroundColor: Colors.white,
      paddingVertical: props.verticalPadding ? props.verticalPadding : 'auto',
      flexDirection: 'row',
      justifyContent: 'space-around',
      alignItems: 'center',
    },
    denyButtonText: {
      ...Buttons.buttonConsentTextBase,
      color: Colors.purpleMain,
    },
    confirmButtonText: {
      ...Buttons.buttonConsentTextBase,
      color: props.disabled ? Colors.blackMain050 : Colors.sandLight,
    },
    denyButton: {
      width: 'auto',
    },
    confirmButton: {
      paddingHorizontal: Spacing.LG,
      borderRadius: 4,
      backgroundColor: props.disabled ? Colors.lightGrey : Colors.purpleMain,
    },
  })

  const {
    disabled,
    denyDisabled,
    denyText,
    handleConfirm,
    handleDeny,
    confirmText,
  } = props

  return (
    <View style={styles.buttonBlock}>
      <Button
        disabled={denyDisabled}
        onPress={handleDeny}
        style={{
          container: styles.denyButton,
          text: styles.denyButtonText,
        }}
        upperCase={false}
        text={denyText}
      />
      <Button
        disabled={disabled}
        onPress={handleConfirm}
        style={{
          container: styles.confirmButton,
          text: styles.confirmButtonText,
        }}
        upperCase={false}
        text={confirmText}
      />
    </View>
  )
}
