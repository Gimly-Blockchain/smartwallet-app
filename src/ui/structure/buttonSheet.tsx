import React from 'react'
import { ActionSheet } from './actionSheet'
import I18n from 'src/locales/i18n'
import { View, StyleSheet } from 'react-native'
import { JolocomButton } from './'
import strings from 'src/locales/strings'
import { fontMedium } from 'src/styles/typography'

const styles = StyleSheet.create({
  wrapper: {
    flexDirection: 'row',
  },
  confirm: {
    flex: 2,
  },
  cancel: {
    flex: 1,
  },
  confirmText: {
    fontFamily: fontMedium,
  },
})

interface Props {
  show?: boolean
  onConfirm: () => void
  onCancel: () => void
  disabledConfirm?: boolean
}

export const ButtonSheet = ({
  show = true,
  onConfirm,
  onCancel,
  disabledConfirm,
}: Props) => {
  return (
    <ActionSheet showSlide={show}>
      <View style={styles.wrapper}>
        <JolocomButton
          textStyle={styles.confirmText}
          containerStyle={styles.confirm}
          disabled={!!disabledConfirm}
          onPress={onConfirm}
          text={I18n.t(strings.SAVE)}
        />
        <JolocomButton
          containerStyle={styles.cancel}
          onPress={onCancel}
          text={I18n.t(strings.CANCEL)}
          transparent
        />
      </View>
    </ActionSheet>
  )
}
