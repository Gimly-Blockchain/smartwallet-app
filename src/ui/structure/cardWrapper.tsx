import React, { ReactNode } from 'react'
import { View, StyleSheet, Text } from 'react-native'
import { Spacing, Colors, Typography } from '../../styles'

const styles = StyleSheet.create({
  card: {
    flexDirection: 'row',
    padding: Spacing.MD,
    backgroundColor: Colors.white,
    borderColor: Colors.lightGrey,
    borderBottomWidth: 1,
  },
  leftIconSection: {
    paddingHorizontal: Spacing.XS,
  },
  rightSection: {
    flex: 1,
    marginLeft: Spacing.LG,
  },
  mainInfo: {
    ...Typography.cardMainText,
    marginTop: Spacing.SM,
  },
})

/**
 * |------------------------------------------------------|
 * | left  |             title                            |
 * | Icon  |----------------------------------------------|
 * |       |                                              |
 * |       |            children                          |
 * |       |                                              |
 * |       |                                              |
 * |       |                                              |
 * |-------|----------------------------------------------|
 */

interface CardWrapperProps {
  leftIcon?: ReactNode
  title: string
}

export const CardWrapper: React.FC<CardWrapperProps> = (props): JSX.Element => (
  <View style={styles.card}>
    {props.leftIcon && (
      <View style={styles.leftIconSection}>{props.leftIcon}</View>
    )}
    <View style={styles.rightSection}>
      <Text style={Typography.cardSecondaryTextBlack}>{props.title}:</Text>
      {props.children}
    </View>
  </View>
)
