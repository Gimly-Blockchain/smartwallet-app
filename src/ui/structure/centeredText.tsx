import React from 'react'
import { StyleSheet, TextStyle, Text, StyleProp } from 'react-native'

const styles = StyleSheet.create({
  text: {
    width: '100%',
    textAlign: 'center' as 'center',
    fontSize: 16,
  },
})

interface Props {
  msg: string
  style?: StyleProp<TextStyle>
}

export const CenteredText: React.SFC<Props> = props => (
  <Text style={[styles.text, props.style]}>{props.msg}</Text>
)
