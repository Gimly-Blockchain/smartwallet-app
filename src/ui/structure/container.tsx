import React, { ReactNode } from 'react'
import { StyleSheet, ViewStyle, StyleProp, SafeAreaView } from 'react-native'

/**
 * Container
 *
 * The main use for this component is to have a full height and width component
 * that can be used in different components. It should be minimal in its style
 * so as to be reusable without constant overriding of its defaults.
 */

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    height: '100%',
    width: '100%',
  },
})

interface Props {
  testID?: string
  children: ReactNode
  style?: StyleProp<ViewStyle>
}

export const Container: React.FC<Props> = props => (
  <SafeAreaView testID={props.testID} style={[styles.container, props.style]}>
    {props.children}
  </SafeAreaView>
)
