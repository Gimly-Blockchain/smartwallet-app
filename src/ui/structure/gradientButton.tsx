import React from 'react'
import {
  Platform,
  StyleSheet,
  Text,
  TextStyle,
  TouchableOpacity,
  ViewStyle,
} from 'react-native'
import LinearGradient from 'react-native-linear-gradient'
import { fontMain } from '../../styles/typography'
import { white } from '../../styles/colors'

const styles = StyleSheet.create({
  buttonContainer: {
    borderRadius: 8,
    maxWidth: '100%',
    height: 56,
  },
  buttonText: {
    fontFamily: fontMain,
    fontSize: 20,
    color: white,
    fontWeight: 'normal',
    paddingTop: Platform.select({
      ios: 5,
      android: 0,
    }),
  },
})

interface Props {
  onPress: () => void
  text: string
  containerStyle?: ViewStyle
  textStyle? : TextStyle
}

export const GradientButton = (props: Props) => {
  const { onPress, containerStyle, text, textStyle } = props
  return (
    <TouchableOpacity
      style={{ ...styles.buttonContainer, ...containerStyle }}
      activeOpacity={0.8}
      onPress={onPress}
    >
      <LinearGradient
        colors={['rgb(145, 25, 66)', 'rgb(210, 45, 105)']}
        start={{ x: 0, y: 0 }}
        end={{ x: 1, y: 0 }}
        style={{
          borderRadius: 8,
          width: '100%',
          height: '100%',
          alignItems: 'center',
          justifyContent: 'center',
        }}
      >
        <Text style={{ ...styles.buttonText, ...textStyle }}>{text}</Text>
      </LinearGradient>
    </TouchableOpacity>
  )
}
