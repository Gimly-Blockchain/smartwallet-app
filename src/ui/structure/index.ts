export { Container } from './container'
export { CardWrapper } from './cardWrapper'
export { JolocomButton } from './jolocomButton'
