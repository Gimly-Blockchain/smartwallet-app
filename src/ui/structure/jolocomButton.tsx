import React from 'react'
import {
  StyleSheet,
  Text,
  TouchableOpacity,
  ViewStyle,
  TextStyle,
} from 'react-native'
import { Buttons } from 'src/styles'
import LinearGradient from 'react-native-linear-gradient'
import { fontMain } from '../../styles/typography'

const styles = StyleSheet.create({
  container: {
    ...Buttons.buttonStandardContainer,
  },
  text: {
    ...Buttons.buttonStandardText,
    fontFamily: fontMain,
  },
  disabledText: {
    opacity: 0.25,
  },
  gradientWrapper: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: 26,
  },
})

interface Props {
  onPress: () => void
  text: string
  disabled?: boolean
  containerStyle?: ViewStyle
  textStyle?: TextStyle
  testID?: string
  transparent?: boolean
}

export const JolocomButton: React.FC<Props> = props => {
  const {
    onPress,
    containerStyle,
    textStyle,
    text,
    disabled,
    transparent,
    testID,
  } = props
  const onButtonPress = () => (disabled ? null : onPress())
  const gradientColors = ['rgb(145, 25, 66)', 'rgb(210, 45, 105)']
  const gradient = transparent ? ['transparent', 'transparent'] : gradientColors

  return (
    <LinearGradient
      colors={gradient}
      start={{ x: 0, y: 0 }}
      end={{ x: 1, y: 0 }}
      style={{
        ...styles.container,
        ...containerStyle,
      }}
    >
      <TouchableOpacity
        activeOpacity={0.8}
        onPress={onButtonPress}
        testID={testID}
        style={styles.gradientWrapper}
      >
        <Text style={[styles.text, textStyle, disabled && styles.disabledText]}>
          {text}
        </Text>
      </TouchableOpacity>
    </LinearGradient>
  )
}
