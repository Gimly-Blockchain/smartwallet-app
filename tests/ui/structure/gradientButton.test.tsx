import React from 'react'
import { shallow } from 'enzyme'
import { GradientButton } from '../../../src/ui/structure/gradientButton'
import { TouchableOpacity } from 'react-native'

describe('GradientButton Component', () => {
  const defaultProps = {
    onPress: jest.fn(),
    text: 'Test',
  }

  it('matches snapshot on initial render', () => {
    const component = shallow(<GradientButton {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })

  it('calls onPress when pressed', () => {
    const component = shallow(<GradientButton {...defaultProps} />)
    component.find(TouchableOpacity).simulate('press')
    expect(defaultProps.onPress).toBeCalledTimes(1)
  })

  it('renders the button with custom styles', () => {
    const props = {
      ...defaultProps,
      containerStyle: { marginHorizontal: 40, height: 30 },
    }
    const component = shallow(<GradientButton {...props} />)
    expect(component).toMatchSnapshot()
  })
})
