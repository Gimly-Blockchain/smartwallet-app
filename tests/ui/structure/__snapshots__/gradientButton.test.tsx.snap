// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GradientButton Component matches snapshot on initial render 1`] = `
<TouchableOpacity
  activeOpacity={0.8}
  onPress={[MockFunction]}
  style={
    Object {
      "borderRadius": 8,
      "height": 56,
      "maxWidth": "100%",
    }
  }
>
  <LinearGradient
    colors={
      Array [
        "rgb(145, 25, 66)",
        "rgb(210, 45, 105)",
      ]
    }
    end={
      Object {
        "x": 1,
        "y": 0,
      }
    }
    start={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 8,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "TT Commons",
          "fontSize": 20,
          "fontWeight": "normal",
          "paddingTop": 5,
        }
      }
    >
      Test
    </Text>
  </LinearGradient>
</TouchableOpacity>
`;

exports[`GradientButton Component renders the button with custom styles 1`] = `
<TouchableOpacity
  activeOpacity={0.8}
  onPress={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  style={
    Object {
      "borderRadius": 8,
      "height": 30,
      "marginHorizontal": 40,
      "maxWidth": "100%",
    }
  }
>
  <LinearGradient
    colors={
      Array [
        "rgb(145, 25, 66)",
        "rgb(210, 45, 105)",
      ]
    }
    end={
      Object {
        "x": 1,
        "y": 0,
      }
    }
    start={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 8,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "TT Commons",
          "fontSize": 20,
          "fontWeight": "normal",
          "paddingTop": 5,
        }
      }
    >
      Test
    </Text>
  </LinearGradient>
</TouchableOpacity>
`;
